core/sync.js

```diff
@@ -349,8 +349,8 @@ class Sync {
       log.debug({ path: doc.path }, `Applying else for ${doc.docType} change`)
       let old
       try {
-        old = await this.pouch.getPreviousRevAsync(doc._id, rev)
-      } catch (_) {
+        old = await this.pouch.getPreviousRevAsync(doc._id, doc.sides.rev - rev)
```

Comment:
We don't care what's the document's `_rev` value here. But this is the mechanism Pouch uses to mark revisions and we have to deal with it.
We're lucky the short rev is incremental since it gives us a way to fetch the previous nth revision.